### PR TITLE
Panel view legacy duplicate

### PR DESCRIPTION
**Pull request by pjfitzgibbons**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .cypress/integration/3_panels.spec.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ describe('Testing panels table', () => {
     });
   });
 
-  describe('with a Legacy Panel', () => {
+  describe.only('with a Legacy Panel', () => {
```
Comment (derek-ho) [inline .cypress/integration/3_panels.spec.ts:107]: don't commit this change?

File: .cypress/integration/3_panels.spec.ts [hunk 2]
```diff
@@ -123,18 +123,17 @@ describe('Testing panels table', () => {
       expectUuid(duplicate);
     });
 
-    it('Renames the legacy panel', () => {
-      createLegacyPanel();
-      cy.reload();
-      const cell = cy.get('.euiTableCellContent');
+    it.only('Renames the legacy panel', () => {
```
Comment (derek-ho) [inline .cypress/integration/3_panels.spec.ts:126]: same comment here

File: public/components/custom_panels/custom_panel_view_so.tsx [hunk 3]
```diff
@@ -227,7 +224,7 @@ export const CustomPanelViewSO = (props: CustomPanelViewProps) => {
 
   const renameCustomPanel = (editedCustomPanelName: string) => {
     if (!isNameValid(editedCustomPanelName)) {
-      setToast('Invalid Custom Panel name', 'danger');
+      dispatch(addToast({ title: 'Invalid Custom Panel name', color: 'danger' } as Toast));
```
Comment (derek-ho) [inline public/components/custom_panels/custom_panel_view_so.tsx:227]: setToast?

**Review by mengweieric — APPROVED** [review]

File: public/components/custom_panels/helpers/utils.tsx [hunk 4]
```diff
@@ -417,7 +417,7 @@ export const parseSavedVisualizations = (
 export const isDateValid = (
   start: string | Moment | undefined,
   end: string | Moment | undefined,
-  setToast: (
+  setToast?: (
```
Comment (joshuali925) [inline public/components/custom_panels/helpers/utils.tsx:420]: when is this optional?

File: public/components/custom_panels/helpers/utils.tsx [hunk 5]
```diff
@@ -445,22 +445,22 @@ const checkWhereClauseExists = (query: string) => {
...
-    setToast('Please remove index from PPL Filter', 'danger', undefined);
-    return false;
+    setToast && setToast('Please remove index from PPL Filter', 'danger', undefined);
+    return [false, 'Please remove index from PPL Filter'];
   }
   if (query && !checkWhereClauseExists(query)) {
-    setToast('PPL filters should start with a where clause', 'danger', undefined);
-    return false;
+    setToast && setToast('PPL filters should start with a where clause', 'danger', undefined);
+    return [false, 'PPL filters should start with a where clause'];
```
Comment (joshuali925) [inline public/components/custom_panels/helpers/utils.tsx:461]: n.p. strings are used twice, extract as constants?

File: public/components/custom_panels/home.tsx [hunk 6]
```diff
@@ -93,8 +101,10 @@ export const Home = ({
 
   const setToast = (title: string, color = 'success', text?: ReactChild, side?: string) => {
     if (!text) text = '';
-    setToastRightSide(!side ? true : false);
-    setToasts([...toasts, { id: new Date().toISOString(), title, text, color } as Toast]);
+    dispatch(setToastRightSide(side !== 'left'));
+    dispatch(addToast({ id: new Date().toISOString(), title, text, color } as Toast));
+    // setToastRightSide(!side ? true : false);
+    // setToasts([...toasts, { id: new Date().toISOString(), title, text, color } as Toast]);
```
Comment (joshuali925) [inline public/components/custom_panels/home.tsx:107]: remove if not used?

File: public/components/custom_panels/redux/panel_slice.ts [hunk 7]
```diff
@@ -206,6 +243,32 @@ export const renameCustomPanel = (editedCustomPanelName: string, id: string) =>
...
 
+export const addVisualizationToCurrentPanel = ({
+  savedVisualizationId,
+  oldVisualizationId,
+}: {
+  savedVisualizationId: string;
+  oldVisualizationId?: string;
+}) => (dispatch, getState) => {
+  const panel = getState().customPanel.panel;
+  console.log('addvis ', { panel });
```
Comment (joshuali925) [inline public/components/custom_panels/redux/panel_slice.ts:254]: remove debugging lines

File: .cypress/integration/3_panels.spec.ts (hunk 1 above)
Comment (pjfitzgibbons) [reply .cypress/integration/3_panels.spec.ts:107]: Will Remove

**Pull request closed without merging**